lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -2,11 +2,43 @@
 
 module Reports
   class SiegeReport
+
+    attr_reader :building
+
     def initialize(building:)
+      @building = building
     end
 
     def call
-      raise NotImprementedYet
+      return 0 unless no_soliders?
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś wyżej, żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)